### PR TITLE
Allow Adding and Editing Library Creation Date

### PR DESCRIPTION
**Pull request by Kashif-S**
Jira: GLT-2419

### CODE REVIEW

File: miso-web/src/main/webapp/scripts/hot_library.js [hunk 1]
```diff
@@ -292,6 +292,29 @@ HotTarget.library = (function() {
...
+            header: 'Date of creation',
+            data: 'creationDate',
+            type: 'date',
+            dateFormat: 'YYYY-MM-DD',
+            datePickerConfig: {
+              firstDay: 0,
+              numberOfMonths: 1
+            },
+            allowEmpty: true,
+            include: true,
```
Comment (callunity) [inline miso-web/src/main/webapp/scripts/hot_library.js]: `!config.isLibraryReceipt`

Comment (callunity) [reply miso-web/src/main/webapp/scripts/hot_library.js]: Hmm. I think it should be null as we don't necessarily have that information available (as the libraries would have been created at the "sending" lab, not in our lab).

**Comment by sqwci** [comment]
SonarQube analysis reported 5 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [HibernateLibraryDao.java#L289](https://github.com/TGAC/miso-lims/blob/15fc3d721fad75c35727b3d18d20a5f7f8e57d4b/sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateLibraryDao.java#L289): Define a constant instead of duplicating this literal "sample" 5 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [hot_library.js#L69](https://github.com/TGAC/miso-lims/blob/15fc3d721fad75c35727b3d18d20a5f7f8e57d4b/miso-web/src/main/webapp/scripts/hot_library.js#L69): Rename "pt" as this name is already used in declaration at line 48. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)
1. ![MAJOR][MAJOR] [hot_library.js#L73](https://github.com/TGAC/miso-lims/blob/15fc3d721fad75c35727b3d18d20a5f7f8e57d4b/miso-web/src/main/webapp/scripts/hot_library.js#L73): Rename "indices" as this name is already used in declaration at line 52. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)
1. ![MAJOR][MAJOR] [hot_library.js#L83](https://github.com/TGAC/miso-lims/blob/15fc3d721fad75c35727b3d18d20a5f7f8e57d4b/miso-web/src/main/webapp/scripts/hot_library.js#L83): Rename "pt" as this name is already used in declaration at line 48. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)
1. ![MINOR][MINOR] [HibernateLibraryDao.java#L289](https://github.com/TGAC/miso-lims/blob/15fc3d721fad75c35727b3d18d20a5f7f8e57d4b/sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateLibraryDao.java#L289): Reorder the modifiers to comply with the Java Language Specification. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AModifiersOrderCheck)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by sqwci** [comment]
SonarQube analysis reported 5 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [HibernateLibraryDao.java#L289](https://github.com/TGAC/miso-lims/blob/0c7ed1d3c355baffdbca8d25c1484da048267f40/sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateLibraryDao.java#L289): Define a constant instead of duplicating this literal "sample" 5 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [hot_library.js#L69](https://github.com/TGAC/miso-lims/blob/0c7ed1d3c355baffdbca8d25c1484da048267f40/miso-web/src/main/webapp/scripts/hot_library.js#L69): Rename "pt" as this name is already used in declaration at line 48. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)
1. ![MAJOR][MAJOR] [hot_library.js#L73](https://github.com/TGAC/miso-lims/blob/0c7ed1d3c355baffdbca8d25c1484da048267f40/miso-web/src/main/webapp/scripts/hot_library.js#L73): Rename "indices" as this name is already used in declaration at line 52. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)
1. ![MAJOR][MAJOR] [hot_library.js#L83](https://github.com/TGAC/miso-lims/blob/0c7ed1d3c355baffdbca8d25c1484da048267f40/miso-web/src/main/webapp/scripts/hot_library.js#L83): Rename "pt" as this name is already used in declaration at line 48. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)
1. ![MINOR][MINOR] [HibernateLibraryDao.java#L289](https://github.com/TGAC/miso-lims/blob/0c7ed1d3c355baffdbca8d25c1484da048267f40/sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateLibraryDao.java#L289): Reorder the modifiers to comply with the Java Language Specification. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AModifiersOrderCheck)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by sqwci** [comment]
SonarQube analysis reported 5 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [HibernateLibraryDao.java#L289](https://github.com/TGAC/miso-lims/blob/f1bd07592d3dfac306b0bb068f5fc8dfe6f10489/sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateLibraryDao.java#L289): Define a constant instead of duplicating this literal "sample" 5 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [hot_library.js#L69](https://github.com/TGAC/miso-lims/blob/f1bd07592d3dfac306b0bb068f5fc8dfe6f10489/miso-web/src/main/webapp/scripts/hot_library.js#L69): Rename "pt" as this name is already used in declaration at line 48. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)
1. ![MAJOR][MAJOR] [hot_library.js#L73](https://github.com/TGAC/miso-lims/blob/f1bd07592d3dfac306b0bb068f5fc8dfe6f10489/miso-web/src/main/webapp/scripts/hot_library.js#L73): Rename "indices" as this name is already used in declaration at line 52. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)
1. ![MAJOR][MAJOR] [hot_library.js#L83](https://github.com/TGAC/miso-lims/blob/f1bd07592d3dfac306b0bb068f5fc8dfe6f10489/miso-web/src/main/webapp/scripts/hot_library.js#L83): Rename "pt" as this name is already used in declaration at line 48. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)
1. ![MINOR][MINOR] [HibernateLibraryDao.java#L289](https://github.com/TGAC/miso-lims/blob/f1bd07592d3dfac306b0bb068f5fc8dfe6f10489/sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateLibraryDao.java#L289): Reorder the modifiers to comply with the Java Language Specification. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AModifiersOrderCheck)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**